input/drivers_joypad/switch_joypad.c

```diff
@@ -170,49 +170,117 @@ static void switch_joypad_destroy(void)
 }
 
 #ifdef HAVE_LIBNX
-int lastMode = 0; /* 0 = handheld, 1 = whatever */
+int previousHandheld = -1; /* 1 = handheld, 0 = docked, -1 = first use */
+uint previousSplitJoyconSetting[MAX_USERS] = { 0 };
```

Comment:
Can you make these variables static if you aren't going to use them outside this source file?

Also, can you not use camel case naming conventions? For instance, 'previousHandheld' should be 'previous_handheld', 
'previousSplitJoyconSetting' should be 'previous_split_joycon_setting', and so on.

Reply:
Thanks! Fixed in latest force-push.